deploy/018-native-token-gateway.ts

```diff
@@ -0,0 +1,79 @@
+import { ethers } from "hardhat";
+import { DeployFunction } from "hardhat-deploy/types";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+import { contracts as ilBscMainnet } from "../deployments/bscmainnet.json";
+import { contracts as ilBscTestnet } from "../deployments/bsctestnet.json";
+import { contracts as ilEthereum } from "../deployments/ethereum.json";
+import { contracts as ilSepolia } from "../deployments/sepolia.json";
+import { getConfig } from "../helpers/deploymentConfig";
+
+interface NetworkConfig {
+  [poolName: string]: string;
+}
+
+const VWNativePooL: { [key: string]: NetworkConfig } = {
+  hardhat: {
+    Comptroller_LiquidStakedBNB: ilBscTestnet.VToken_vWBNB_LiquidStakedBNB.address,
+  },
+  bsctestnet: {
+    Comptroller_LiquidStakedBNB: ilBscTestnet.VToken_vWBNB_LiquidStakedBNB.address,
+  },
+  bscmainnet: {
+    Comptroller_LiquidStakedBNB: ilBscMainnet.VToken_vWBNB_LiquidStakedBNB.address,
+  },
+  sepolia: {
+    Comptroller_Core: ilSepolia.VToken_vWETH_Core.address,
+  },
+  ethereum: {
+    Comptroller_Core: ilEthereum.VToken_vWETH_Core.address,
+  },
+};
+
+const getVWNativeToken = (networkName: string, poolName: string): string => {
+  const poolAssociatedVToken = VWNativePooL[networkName];
+  if (poolAssociatedVToken === undefined) {
+    throw new Error(`config for network ${networkName} is not available.`);
+  }
+
+  const vTokenAddress = poolAssociatedVToken[poolName];
+  if (vTokenAddress === undefined) {
+    throw new Error(`config for pool ${poolName} is not available.`);
+  }
+
+  return vTokenAddress;
+};
+
+// Enter the desired pool name here
+const poolName: string = "Comptroller_Core";
```

Comment:
I think we could get this value from the vToken contract, in an automatic way (depending on the final key to be used, see [above](https://github.com/VenusProtocol/isolated-pools/pull/361/files#r1500758111)). This way we won't need to modify the deployment script every time we want to deploy a new NativeTokenGateway

Reply:
https://github.com/VenusProtocol/isolated-pools/pull/361/commits/4211d2dd29d2ca2e64b5349b40a222223e4590ba